src/openrct2/network/NetworkBase.cpp

```diff
@@ -2058,7 +2057,7 @@ NetworkPlayer* NetworkBase::AddPlayer(const std::string& name, const std::string
             }
             else
             {
-                player->Group = networkUser->GroupId.GetValueOrDefault(GetDefaultGroup());
+                player->Group = networkUser->GroupId.has_value() ? *networkUser->GroupId : GetDefaultGroup();
```

Comment:
Btw, this could have been simplified even further as we're now using optional:
```c++
player->Group = networkUser->GroupId.value_or(GetDefaultGroup());
```

Reply:
@tupaschoal Wouldn't that always call `GetDefaultGroup()` while with the ternary operator it would only call it if `has_value()` returns false?

Reply:
```suggestion
                player->Group = networkUser->GroupId.has_value() ? networkUser->GroupId.value() : GetDefaultGroup(); 
```

Reply:
> @tupaschoal Wouldn't that always call `GetDefaultGroup()` while with the ternary operator it would only call it if `has_value()` returns false?

Good point, I think so